ios-xcframework/build.sh

```diff
@@ -97,15 +97,52 @@ do
     -output "$XCFRAMEWORKS_DIR/$CURRENT_FRAMEWORK_NAME.xcframework"
 done
 
-log 'compression' 'Zipping Gutenberg XCFrameworks'
-ZIP_PATH=$XCFRAMEWORKS_DIR/$MAIN_FRAMEWORK_NAME.zip
-zip -rq "$ZIP_PATH" \
-  $XCFRAMEWORKS_DIR/Aztec.xcframework \
-  $XCFRAMEWORKS_DIR/Gutenberg.xcframework \
-  $XCFRAMEWORKS_DIR/React.xcframework \
-  $XCFRAMEWORKS_DIR/RNTAztecView.xcframework \
-  $XCFRAMEWORKS_DIR/yoga.xcframework
-echo "Gutenberg XCFrameworks ZIP generated at $ZIP_PATH"
+log 'bulb' 'Work around "Gutenberg class ahead of Gutenberg module" issue'
+# We generate a Gutenberg framework with a Gutenberg class in it.
+# This results in some types in the swiftinterface being defined as Gutenberg.Gutenberg.Type which cannot be resolved.
+# There is no fix, but we can work around it by post-processing the interface and flattening the namespace where appropriate.
+#
+# See https://developer.apple.com/forums/thread/123253
+#
+# TODO: There has to be a RegEx that can do what this sequence does in one go. Something along the lines of s/(^|\s|\[)Gutenberg\.(?!self)/\1/g but that doesn't throw errors
+find $XCFRAMEWORKS_DIR -name "*.swiftinterface" -exec sed -i -e 's/Gutenberg\.self/PLACEHOLDER_TO_REVERT_SED/g' {} \;
+find $XCFRAMEWORKS_DIR -name "*.swiftinterface" -exec sed -i -e 's/ Gutenberg\./ /g' {} \;
+find $XCFRAMEWORKS_DIR -name "*.swiftinterface" -exec sed -i -e 's/\[Gutenberg\./[/g' {} \;
+find $XCFRAMEWORKS_DIR -name "*.swiftinterface" -exec sed -i -e 's/PLACEHOLDER_TO_REVERT_SED/Gutenberg.self/g' {} \;
+
+log 'compression' 'Compressing Gutenberg XCFrameworks'
+
+# For CocoaPods to find the XCFrameworks in the archive, it needs to have a certain folder structure.
+#
+# See for example what Firebase does,
+# https://github.com/CocoaPods/Specs/blob/master/Specs/e/2/1/FirebaseAnalytics/9.6.0/FirebaseAnalytics.podspec.json
+#
+# Archive/
+#   - Frameworks/
+#     - A.xcframework
+#     - B.xcframework
+#   - dummy.txt
```

Comment:
Note that even though I put all this in place, I'm still unable to get CocoaPods to find the xcframeworks in WordPress iOS 😭

I decided to leave the setup here in the hope to figure out how to make it work in the future